### PR TITLE
[feat] deprecated `return_transform`, enabled 3D augmentations in AugmenationSequential

### LINKED ISSUE #1341: Using a sequence of 3D augmentations in AugmentationSequential
### Describe the bug

I'm using Kornia 3D augmentations in a 3D image segmentation project. I found out that AugmentatonSequence doesn't work for 3D transforms.

Input: A 5D torch tensor made of a batch of 3D volumes in the shape of [B X  C X D X H X W]

Example code to reproduce the bug:
```
import kornia.augmentation as K
from kornia.constants import DataKey

transforms = [
        K.RandomHorizontalFlip3D(p=0.5),
        K.RandomVerticalFlip3D(p=0.5),
        K.RandomRotation3D((30, 10, 10), p=0.5)
    ]
aug_list = K.AugmentationSequential(*transforms, data_keys=[DataKey.INPUT, DataKey.MASK])

img_shape = (1, 1, 4, 16, 16)
img = torch.zeros(img_shape)
mask = torch.zeros(img_shape)
aug_list(img, mask)
```
## Error
1. Fails with the following error. `RuntimeError: input shape expected to be in (2, 4) while got torch.Size([1, 1, 4, 16, 16])`
This is caused by the tensor dimension check `_, out_shape = self.autofill_dim(inp, dim_range=(2, 4))` in the AugmentationSequence [[Line 247](https://github.com/kornia/kornia/blob/9ce2bca334fa599fdcfc6abb07b7e9f8533b7d67/kornia/augmentation/container/augment.py#L247)]
2. Even if I give a single 3D image and its mask as two 4D tensors, the augmentations are not applied to the mask because AugmentationBase3D is not a descendant of GeometricAugmentation2D.

## Questions
* Is there a different way to apply a sequence of 3D augmentations on a batch of 3D images?
* Do we need to have a separate base class for 3D geometric augmentations?

Example:
```
# TODO check for 3D geometric augmentations
elif isinstance(module, GeometricAugmentationBase3D) and dcate==DataKey.MASK:
     input, label = ApplyInverse.apply_by_key(input, label, module, param, dcate)
```

### Reproduction steps

```bash
1. Run the code sample I have mentioned in the above section.
```

### Expected behavior

The AugmentationSequence needs to be applied to the batch of images and masks (5D tensors) successfully.

### Environment

- PyTorch Version (e.g., 3.8): 1.8
- OS (e.g., Linux): Linux
- How you installed PyTorch: `pip`
- Build command you used (if compiling from source):
- Python version: 3.8
- CUDA/cuDNN version: 11.4
- GPU models and configuration:
```

### LINKED ISSUE #1502: Deprecate return_transform in augmentations
We initially keep the `return_transform` for inverting the augmentation. While we implemented the `inverse` function now, I would suggest to deprecate the `return_transform` by moving to:
```python
class RandomAug(...):
    @property
    def transformation_matrix(self):
        return self._transform_mat
```

This can also make the codebase cleaner for handling the output.

### PR DESCRIPTION
**Pull request by shijianjian**
Fixes #1502 

Fixes #1341

### CODE REVIEW

File: kornia/augmentation/base.py [hunk 1]
```diff
@@ -63,14 +63,18 @@ def __repr__(self) -> str:
...
         """Check if a transformation matrix is returned, it has to be in the same batching mode as output."""
         raise NotImplementedError
 
     def transform_tensor(self, input: torch.Tensor) -> torch.Tensor:
         """Standardize input tensors."""
         raise NotImplementedError
 
+    def transform_output_tensor(self, output: torch.Tensor, ori_shape: Tuple) -> torch.Tensor:
+        """Standardize output tensors."""
+        return _transform_output_shape(output, ori_shape) if self.keepdim else output
```
Comment (edgarriba) [inline kornia/augmentation/base.py:76]:
```suggestion
        return _transform_output_shape(output, original_shape) if self.keepdim else output
```

File: kornia/augmentation/base.py [hunk 2]
```diff
@@ -175,9 +177,19 @@ def __init__(
...
         self.return_transform = return_transform
+        self._transform_matrix: torch.Tensor
+        if return_transform is not None:
+            raise ValueError(
+                "`return_transform` is deprecated. Please access the transformation matrix with "
+                "`.transform_matrix`. For chained matrices, please use `AugmentationSequential`.",
+            )
+
+    @property
+    def transform_matrix(self,) -> torch.Tensor:
```
Comment (edgarriba) [inline kornia/augmentation/base.py:188]: Does it return in batch ?

File: kornia/augmentation/container/image.py [hunk 3]
```diff
@@ -240,29 +238,56 @@ def contains_label_operations(self, params: List[ParamItem]) -> bool:
...
         return output
 
-    def get_transformation_matrix(self, input: torch.Tensor, params: Optional[List[ParamItem]] = None) -> torch.Tensor:
-        """Compute the transformation matrix according to the provided parameters."""
+    def get_transformation_matrix(
+        self, input: torch.Tensor, params: Optional[List[ParamItem]] = None, recompute: bool = False
+    ) -> Optional[torch.Tensor]:
+        """Compute the transformation matrix according to the provided parameters.
+
+        args:
```
Comment (edgarriba) [inline kornia/augmentation/container/image.py:253]:
```suggestion
        Args:
```

File: test/augmentation/test_augmentation.py [hunk 4]
```diff
@@ -166,52 +159,12 @@ def _test_smoke_call_implementation(self, params):
...
-        assert final_transformation.shape == expected_transformation_shape
-        assert_close(final_transformation, transformation @ test_transform, atol=1e-4, rtol=1e-4)
-
-        output, final_transformation = augmentation(
-            (test_input, test_transform), params=generated_params, return_transform=True
-        )
-        assert output.shape[0] == batch_shape[0]
-        assert final_transformation.shape == expected_transformation_shape
-        assert_close(final_transformation, transformation @ test_transform, atol=1e-4, rtol=1e-4)
+        assert augmentation._transform_matrix.shape == expected_transformation_shape
```
Comment (edgarriba) [inline test/augmentation/test_augmentation.py:167]:
```suggestion
        assert augmentation.transform_matrix.shape == expected_transformation_shape
```
Better to show with property

**Review by edgarriba — COMMENTED** [review]
LGTM. I like because it simplifies the logic and the api is more clean. Added few nomenclature suggestions. Besides, I believe tutorials and examples must be revisited. One thought, is that we could expose also the samplers at some point in future

File: kornia/augmentation/base.py (hunk 2 above)
Comment (shijianjian) [reply kornia/augmentation/base.py:188]: yes

File: kornia/augmentation/container/augment.py [hunk 5]
```diff
@@ -162,11 +160,20 @@ def __init__(
...
                 warnings.warn("Geometric transformation detected in PatchSeqeuntial, which would break bbox, mask.")
             if isinstance(arg, VideoSequential):
                 self.contains_video_sequential = True
+            # NOTE: only for images are supported for 3D.
+            if isinstance(arg, AugmentationBase3D):
+                self.contains_3d_augmentation = True
+        self._transform_matrix: Optional[torch.Tensor] = None
+
+    @property
+    def transform_matrix(self,) -> Optional[torch.Tensor]:
```
Comment (edgarriba) [inline kornia/augmentation/container/augment.py:175]:
```suggestion
    def transform_matrix(self) -> Optional[Tensor]:
```

**Review by edgarriba — APPROVED** [review]
Still some torch.Tensors but I guess it’s not critical for this PR. It can go in a follow up PR specially integrating the Image API

**Pull request merged**